### PR TITLE
Mnist model and training

### PR DESCRIPTION
**Pull request by Charl-AI**
## Summary of changes made
* Create initial model and trainer to train on MNIST

## Testing
Approx 99% accuracy after 2 epochs

## Code Quality
* [x] Passes static analysis checks
* [x] Code is black formatted
* [ ] Unit tests written and passed

## Checklist
* [ ] Added PR label(s)
* [ ] Added linked issue

### CODE REVIEW

File: src/model/resnet.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    def add_model_specific_args(parent_parser):
+        parser = parent_parser.add_argument_group("StochasticUnetSystem")
+        parser.add_argument(
+            "-lr",
+            "--learning_rate",
+            help="learning rate of optimiser",
+            type=float,
+            default=0.01,
+        )
+        return parent_parser
```
Comment (Charl-AI) [inline src/model/resnet.py:70]: /explain
